programs/stake/src/stake_instruction.rs

```diff
@@ -43,86 +43,75 @@ impl<E> DecodeError<E> for StakeError {
 
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
 pub enum StakeInstruction {
-    /// `Initialize` a stake with Lockup and Authorized information
+    /// Initialize a stake with lockup and authorization information
     ///
-    /// Expects 2 Accounts:
-    ///    0 - Uninitialized StakeAccount
-    ///    1 - Rent sysvar
+    /// # Account references
+    ///   0. [WRITE] Uninitialized stake account
```

Comment:
I like the effort to formalize these instruction definitions.  How about in addition to WRITE and READ we also add SIGNER.  Also, can we assume `READ` and just add `[WRITE]` or `[WRITE, SIGNER]` as appropriate?

Reply:
Yes, I'm running under the assumption that WRITE implies READ.

Reply:
Yeah, can we get rid of the ubiquitous `READ`?

Reply:
Hmm, I see.  But that leaves us with a bunch of docs that say `0. [] Some read-only account`.

Reply:
Here's what it looks like 28d922d